cli/util/fs.rs

```diff
@@ -492,6 +492,68 @@ pub async fn remove_dir_all_if_exists(path: &Path) -> std::io::Result<()> {
   }
 }
 
+mod clone_dir_imp {
```

Comment:
There is a similar optimization for `cp` and `copyFile` APIs. It uses rayon for recursive clonefile and has some other optimizations for small files and fallbacks: https://github.com/denoland/deno/blob/b21004b1d16ad7b67c7b1cd235abf792bf9d9777/ext/fs/std_fs.rs#L511

Maybe could be reused?

Reply:
I'm not sure we should use rayon in non-js sync specific scenarios. I feel like that might make stuff slower by using another thread pool.

Reply:
Gave it a shot anyway, no performance difference but I found it caused a failure when the destination is a directory that already exists - didn't dig into it but seems like something wasn't getting copied over properly (the panic is from setting up the bin entries – the origin file we're trying to symlink doesn't exist)
```
thread 'main' panicked at cli/npm/managed/resolvers/local/bin_entries.rs:89:48:
called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }
```